src/base/utils/fs.cpp

```diff
@@ -118,6 +122,14 @@ bool Utils::Fs::smartRemoveEmptyFolderTree(const QString &path)
         ".DS_Store"
     };
 
+    // deleteFilesList contains unwanted files, usually created by the OS
+    // temp files on linux usually end with '~', e.g. `filename~`
+    const auto predicate = [&deleteFilesList](const QString &f)
+    {
+        return (deleteFilesList.contains(f, Qt::CaseInsensitive)
```

Comment:
now this is slow, deleteFilesList should use a QSet instead.

Reply:
Are you sure? According to [this](https://doc.qt.io/qt-5/containers.html) the lookup for QList is O(1) while the lookup for QSet is amortized O(1). So QList should be faster.

Reply:
>Are you sure? 

Yes I'm sure, I won't bother you with the details as there are many tutorials about algorithm complexity on the net. I will only point out that what you saw O(1) for QList is called *Index* lookup, that is not the same as searching within the container.

Reply:
Also you might want to wrap the entries in deleteFilesList with `QLatin1String()`